File: L6_MaxProdructOfThree.cs
using System;
using System.Collections.Generic;
using System.IO;
using System.Linq;
using System.Text;

public class Solution
{
    public static void Main(string[] args)
    {
        Console.WriteLine(solution(new[] { 0, 1, 1, 1, 7, 8, 1 }));
    }

    public static int solution(int[] A)
    {
        Array.Sort(A);

        var topMaximum = A[A.Length - 1] * A[A.Length - 2] * A[A.Length - 3];
        var bottomMaxium = A[0] * A[1] * A[A.Length - 1];

        return Math.Max(topMaximum, bottomMaxium);
    }
}

// tests

File: L6_MaxProdructOfThree.py
#Max product of 3 in array
def solution(A):
    if len(A) < 3:
        raise Exception("Invalid input")

    A.sort()

    return max(A[0] * A[1] * A[-1], A[-1] * A[-2] * A[-3])

A=[-7,8,9,3,-2,1]
print(solution(A))